utils/NamedType.h

```diff
@@ -0,0 +1,26 @@
+#ifndef NAMEDTYPE
```

Comment:
Let's move this file to common. Then in other files you can include it as `#include "utils/NamedType.h"`